packages/config/src/Config.ts

```diff
@@ -0,0 +1,666 @@
+                    } catch (e) {
+                        throw new ImperativeError({ msg: `An error was encountered while trying to read the file '${layer.path}'.` +
+                            `\nError details: ${e.message}`,
+                                                    suppressDump: true });
+                    }
+                    try {
+                        layer.properties = JSONC.parse(fileContents.toString());
+                        layer.exists = true;
+                    } catch (e) {
+                        throw new ImperativeError({ msg: `Error parsing JSON in the file '${layer.path}'.\n` +
+                            `Please check this configuration file for errors.\nError details: ${e.message}\nLine ${e.line}, Column ${e.column}`,
+                                                    suppressDump: true});
+                    }
+                }
+
+                // Find the active layer
+                if (setActive && layer.exists) {
+                    _._active.user = layer.user;
+                    _._active.global = layer.global;
+                    setActive = false;
+                }
+
+                // Populate any undefined defaults
+                layer.properties.defaults = layer.properties.defaults || {};
+                layer.properties.profiles = layer.properties.profiles || {};
+                layer.properties.plugins = layer.properties.plugins || [];
+                layer.properties.secure = layer.properties.secure || [];
+            });
+        } catch (e) {
+            if (e instanceof ImperativeError) {
+                throw e;
+            } else {
+                throw new ImperativeError({ msg: `An unexpected error occurred during config load: ${e.message}` });
+            }
+        }
+
+        ////////////////////////////////////////////////////////////////////////
+        // load secure fields
+        await _.secureLoad();
+
+        ////////////////////////////////////////////////////////////////////////
+        // Complete
+        return _;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+    // APIs
+    ////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+
+    get api() {
+        // tslint:disable-next-line
+        const outer = this;
+
+        return new class {
+
+            ////////////////////////////////////////////////////////////////////
+            ////////////////////////////////////////////////////////////////////
+            // Profiles API
+            ////////////////////////////////////////////////////////////////////
+            ////////////////////////////////////////////////////////////////////
+
+            // tslint:disable-next-line
+            public profiles = new class {
```

Comment:
there are several tslint rules disabled - is it for inner classes?  would it be "better" to have several, separate class files for ease of testing?